frame/babe/src/lib.rs

```diff
@@ -482,13 +482,13 @@ impl<T: Trait> Module<T> {
 				// place the VRF output into the `Initialized` storage item
 				// and it'll be put onto the under-construction randomness
 				// later, once we've decided which epoch this block is in.
-				Some(primary.vrf_output)
+				Some(primary.randomness)
```

Comment:
can we use `vrf_output.make_bytes` here instead of adding the extra field + check to pre-runtime?

Reply:
I tacitly assumed it'd be done that way in #5785 
```
let pk = ..block_producer_babe_key..;
let t = sc_consensus_babe::authorship::make_transcript( epoch.randomness, slot_number, epoch );
let randomness = primary.vrf_output.attach_input_hash(pk,t)
```

Reply:
Could someone explain how to get those four input values here maybe?  I do not see where block_producer_babe_key lives

Reply:
We don't have access to those keys from the runtime, so if I'm not mistaken we really need to create this outside of the runtime as @sorpaas did.

Reply:
It's only the public babe session key required here, not any private key.  It protects validators who mistakenly create session keys using HDKD.  

There is nothing wrong with passing this in like @sorpaas did either of course, whatever seems simplest.  :)

Reply:
I guess the advantage of doing it on the runtime is that it won't require any explicit migration, i.e. this is implicitly done on the next runtime upgrade. We should be able to get the `AuthorityId` by fetching the authority index from the predigest and then fetching the public session key from the module's storage.

Reply:
Yeah, that was my reasoning behind asking the question. The approach of computing the VRF randomness outside and adding an extra field to the header will require synchronized runtime/node migrations via `BabeApi +  RuntimeApiVersion` which sounds more annoying.

Reply:
That said, light clients (which should do this entire randomness beacon calculation _outside of the runtime_) would need similar migration logic.